### PR TITLE
EVG-2005 allow aborting in-progress tasks when unscheduling build

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/lifecycle.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func SetVersionActivation(versionId string, active bool, caller string) error {
 
 // SetBuildActivation updates the "active" state of this build and all associated tasks.
 // It also updates the task cache for the build document.
-func SetBuildActivation(buildId string, active bool, caller string) error {
+func SetBuildActivation(buildId string, active bool, caller string, abortInProgress bool) error {
```
Comment (tychoish) [inline model/lifecycle.go]: I might be inclined to not modify this interface, and instead just call both methods from the dbconnector? or have two dbconnctor methods that the route calls? 

I worry about the implications of having two queries in one model method, particularly wrt to atomicity implications, and it's clearer, maybe to have them seperate. Also these are sort of the gnarliest parts of the system, and therefore the riskyest, so breaking them apart more is probably the goal.

File: service/build.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func (uis *UIServer) modifyBuild(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	case "set_active":
-		err := model.SetBuildActivation(projCtx.Build.Id, putParams.Active, user.Id)
+		err := model.SetBuildActivation(projCtx.Build.Id, putParams.Active, user.Id, putParams.Abort)
```
Comment (tychoish) [inline service/build.go]: shouldn't this always be false?

File: model/lifecycle.go (hunk 1 above)
Comment (john-m-liu) [reply model/lifecycle.go]: I split these up in the route

File: service/build.go (hunk 2 above)
Comment (john-m-liu) [reply service/build.go]: It should be what the parameters are. The set_active is a bit deceptive because this can both schedule or unschedule the build

**Pull request merged**